mavenembedder-workitem/src/main/java/org/jbpm/process/workitem/mavenembedder/MavenEmbedderWorkItemHandler.java

```diff
@@ -0,0 +1,121 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jbpm.process.workitem.mavenembedder;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.maven.cli.MavenCli;
+import org.jbpm.process.workitem.core.AbstractLogOrThrowWorkItemHandler;
+import org.jbpm.process.workitem.core.util.RequiredParameterValidator;
+import org.jbpm.process.workitem.core.util.Wid;
+import org.jbpm.process.workitem.core.util.WidMavenDepends;
+import org.jbpm.process.workitem.core.util.WidParameter;
+import org.jbpm.process.workitem.core.util.WidResult;
+import org.jbpm.process.workitem.core.util.service.WidAction;
+import org.jbpm.process.workitem.core.util.service.WidService;
+import org.kie.api.runtime.process.WorkItem;
+import org.kie.api.runtime.process.WorkItemManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Wid(widfile = "MavenEmbedderDefinitions.wid", name = "MavenEmbedder",
+        displayName = "MavenEmbedder",
+        defaultHandler = "mvel: new org.jbpm.process.workitem.mavenembedder.MavenEmbedderWorkItemHandler()",
+        documentation = "${artifactId}/index.html",
+        parameters = {
+                @WidParameter(name = "Commands", required = true),
+                @WidParameter(name = "WorkDirectory", required = true),
+                @WidParameter(name = "ProjectRoot")
+        },
+        results = {
+                @WidResult(name = "MavenResults")
+        },
+        mavenDepends = {
+                @WidMavenDepends(group = "${groupId}", artifact = "${artifactId}", version = "${version}")
+        },
+        serviceInfo = @WidService(category = "${name}", description = "${description}",
+                keywords = "maven,mvn,build,execute,pom,project,intall",
+                action = @WidAction(title = "Execute Maven commands")
+        ))
+public class MavenEmbedderWorkItemHandler extends AbstractLogOrThrowWorkItemHandler {
+
+    private static final Logger logger = LoggerFactory.getLogger(MavenEmbedderWorkItemHandler.class);
+    private static final String RESULTS_VALUES = "MavenResults";
+
+    public void executeWorkItem(WorkItem workItem,
+                                WorkItemManager workItemManager) {
+
+        try {
+
+            RequiredParameterValidator.validate(this.getClass(),
+                                                workItem);
+
+            String commands = (String) workItem.getParameter("Commands");
+            String workDir = (String) workItem.getParameter("WorkDirectory");
+            String projectRoot = (String) workItem.getParameter("ProjectRoot");
+            if(projectRoot == null || projectRoot.length() < 1) {
+                projectRoot = "$M2_HOME";
```

Comment:
why setting project root to maven home?

Reply:
this is where .mvn can be found if needed.

Reply:
i will remove this since projectRoot is going to be passed in required field